crates/oxc_linter/src/rules/import/no_named_as_default_member.rs

```diff
@@ -0,0 +1,166 @@
+#![allow(clippy::significant_drop_tightening)]
+use std::collections::HashMap;
+
+use dashmap::mapref::one::Ref;
+use oxc_ast::{
+    ast::{BindingPatternKind, Expression, MemberExpression},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+use oxc_syntax::module_record::ImportImportName;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-import(no-named-as-default-member): {1:?} also has a named export {2:?}")]
+#[diagnostic(severity(warning), help("Check if you meant to write `import {{{2:}}} from {3:?}`"))]
+struct NoNamedAsDefaultMemberDignostic(#[label] pub Span, String, String, String);
+
+/// <https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-as-default-member.md>
+#[derive(Debug, Default, Clone)]
+pub struct NoNamedAsDefaultMember;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Reports use of an exported name as a property on the default export.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // ./bar.js
+    /// export function bar() { return null }
+    /// export default () => { return 1 }
+    ///
+    /// // ./foo.js
+    /// import bar from './bar'
+    /// const bar = foo.bar // trying to access named export via default
+    /// ```
+    NoNamedAsDefaultMember,
+    nursery
+);
+
+impl Rule for NoNamedAsDefaultMember {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let module_record = ctx.semantic().module_record();
+
+        let mut has_members_map: HashMap<&Atom, (Ref<'_, Atom, _, _>, String)> = HashMap::default();
+        for import_entry in &module_record.import_entries {
+            let ImportImportName::Default(_) = import_entry.import_name else {
+                continue;
+            };
+
+            let specifier = import_entry.module_request.name();
+            let Some(remote_module_record_ref) = module_record.loaded_modules.get(specifier) else {
+                continue;
+            };
+
+            if !remote_module_record_ref.exported_bindings.is_empty() {
+                has_members_map.insert(
+                    import_entry.local_name.name(),
+                    (remote_module_record_ref, import_entry.module_request.name().to_string()),
```

Comment:
Is the allocation of `to_string` need to be avoided?

Reply:
Started to use Atom, which is wrapper around `CompactString`, seems to be it will be more memory efficient.